fmtrequest -> abstractfmtrequest (#18935)
pantsbuild/pants: docs/markdown/Writing Plugins/common-plugin-tasks/plugins-fmt-goal.md

    sources: ShellSourceField
```

> Then, hook this up to a new subclass of `AbstractFmtRequest`.

Reviewer: Oops, this one's my bad:
```suggestion
Then, hook this up to a new subclass of `FmtTargetsRequest`.
```

```python

> from pants.core.goals.fmt import AbstractFmtRequest

Reviewer: Also my bad:
```suggestion
from pants.core.goals.fmt import FmtTargetsRequest
```

    )

    result = await Get(ProcessResult, Process, process)

>     return await AbstractFmtResult.create(request, result, output_snapshot)

Reviewer: This line is renaming `FmtResult`, not `FmtRequest`.
```suggestion
    return await FmtResult.create(request, result, output_snapshot)
```

    return await FmtResult.create(request, result, output_snapshot)
```

> The `AbstractFmtRequest.Batch` has `.snapshot`, which stores the list of files and the `Digest` for each source file.

Reviewer: Let's blame me on this one too...

```suggestion
The `ShfmtRequest.Batch` object has `.snapshot`, which stores the list of files and the `Digest` for each source file.
```
